### ISSUE TITLE
Should we let the `StructureMatcher` method work at the affine space group classification level instead of the crystallographic space group classification level?

### ISSUE THREAD

**Issue by hongyi-zhao** [materialsproject/pymatgen#3612]
See my following testing:

```python
$ python ./pymatgenStructureMatcher.py 5/POSCAR 6/POSCAR -s0.53 -l 0.01 -a5
Structures are similar: True
Transformation Information:
(array([[ 1,  0,  0],
       [ 0, -1,  0],
       [ 0,  1,  1]]), array([-4.07660017e-17, -4.16198382e-01,  3.28125000e-01]), [3, 58, 59, 60, 33, 8, 0, 1, 56, 12, 13, 68, 25, 70, 71, 63, 10, 11, 21, 22, 23, 24, 15, 44, 36, 37, 20, 30, 49, 32, 7, 34, 35, 27, 28, 47, 48, 40, 41, 42, 43, 26, 18, 19, 46, 39, 31, 50, 51, 52, 53, 45, 38, 29, 64, 4, 5, 6, 61, 62, 54, 55, 2, 66, 67, 14, 69, 16, 17, 9, 57, 65, 75, 102, 77, 78, 79, 80, 72, 73, 101, 93, 85, 86, 87, 88, 98, 81, 91, 83, 84, 94, 95, 96, 97, 89, 90, 82, 92, 76, 103, 104, 105, 106, 107, 99, 100, 74, 144, 145, 146, 138, 139, 140, 147, 148, 149, 141, 142, 143, 123, 124, 125, 108, 109, 110, 126, 127, 128, 111, 112, 113, 129, 130, 131, 114, 115, 116, 117, 118, 119, 132, 133, 134, 120, 121, 122, 135, 136, 137, 186, 187, 188, 174, 175, 176, 189, 190, 191, 177, 178, 179, 159, 160, 161, 150, 151, 152, 153, 154, 155, 162, 163, 164, 156, 157, 158, 165, 166, 167, 180, 181, 182, 168, 169, 170, 183, 184, 185, 171, 172, 173])
```
As you can see, according to the [Classification systems](https://en.wikipedia.org/wiki/Space_group#Classification_systems) of the space group, the `StructureMatcher` method implemented currently works at the affine space group classification level instead of the crystallographic space group classification level, aka, it permits the non-proper rotation which doesn't preserve [orientation](https://en.wikipedia.org/wiki/Orientation_(mathematics)).

But as Bernd pointed out [here](https://www.math.ru.nl/~souvi/krist_09/cryst.pdf) on page 35, as shown below:
> Since crystals occur in physical space and physical space can only be transformed by orientation preserving mappings, space groups are only regarded as equivalent if they are conjugate by an orientation preserving affine mapping, i.e. by an affine mapping that has linear part with positive determinant.

![image](https://github.com/materialsproject/pymatgen/assets/11155854/b722214c-1c39-4995-9a01-44d328d2c397)

So, should we let the `StructureMatcher` method work at the affine space group classification level instead of the crystallographic space group classification level?

Attached are all the test files mentioned above. Please check them.

[test_StructureMatcher.zip](https://github.com/materialsproject/pymatgen/files/14211296/test_StructureMatcher.zip)

Regards,
Zhao

**Comment by janosh** [comment]
this sounds interesting and i can see why you might want to allow only orientation-preserving rotations. but i don't have enough background to say why the current behavior was chosen in the first place. potentially there were reasons i'm unaware of. hoping others with more context can chime in

**Comment by mkhorton** [comment]
Is this the same issue encountered [here](http://aflowlib.org/prototype-encyclopedia/enantiomorph_info.html) with judging equivalence of structures with enantiomorphic space groups?

**Comment by hongyi-zhao** [comment]
@mkhorton Yes. Exactly in 3-dimension. But in higher dimensions, this is a more complicated problem to solve and I have given a [gap implementation](https://github.com/hongyi-zhao/CrystKit/blob/e4816b6292ad11361342115b560aa4fdd175bb7f/gap/CrystKit.gi#L1055) based on [the algorithm](https://doi.org/10.1107/S0108767303004161) suggested by Bernd.

**Comment by wladerer** [comment]
I think it is best to consider that once you get to this kind of granularity of equivalence, the deviations in total energies may be below the threshold of chemical accuracy [1](https://doi.org/10.1016/0375-9601(79)90647-9) [2](https://www.pnas.org/doi/full/10.1073/pnas.96.8.4252)- even by the standards of the highest levels of theory. 

With that said, there are in fact very important observable differences in chiral crystals. Circular dichroism and (reciprocal space) topology being two properties that are sensitive to such symmetries. 

Solving this problem is incredibly interesting, but I am unsure how much it will benefit the users of pymatgen, especially if this  routine incurs additional additional computational overhead. And naturally, this new feature will require additional maintenance that requires a niche expertise. 

I think the best thing, which might actually help address #2593 , is to diversify the routines offered in StructureMatcher ? It would allow for the users to determine what degree of specificity they really care about and perhaps that can reduce the computational load if they are looking for a course grain matching. 

But I suppose we have to wonder if readily available solutions exist elsewhere, then does pymatgen bear this responsibility?
